File: API/Features/Sales/Invoices/Dtos/Read/InvoiceReadDto.cs
using System;
using System.Collections.Generic;
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Sales.Invoices {

    public class InvoiceReadDto : IMetadata {

        // PK
        public Guid InvoiceId { get; set; }
        // Fields
        public DateTime Date { get; set; }
        public DateTime TripDate { get; set; }
        public int InvoiceNo { get; set; }
        public decimal NetAmount { get; set; }
        public decimal VatPercent { get; set; }
        public decimal VatAmount { get; set; }
        public decimal GrossAmount { get; set; }
        public string Remarks { get; set; }
        public bool IsEmailSent { get; set; }
        public bool IsCancelled { get; set; }
        // Child tables
        public InvoiceFormAadeVM Aade { get; set; }
        public List<InvoicePortReadDto> InvoicesPorts { get; set; }
        // Navigation
        public SimpleEntity Customer { get; set; }
        public SimpleEntity Destination { get; set; }
        public DocumentTypeVM DocumentType { get; set; }
        public SimpleEntity PaymentMethod { get; set; }
        public SimpleEntity Ship { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/sales/invoices/classes/dtos/form/invoice-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { AadeVM } from '../../view-models/form/aade-vm'
import { DocumentTypeVM } from '../../view-models/form/documentType-vm'
import { InvoiceXmlPartyTypeDto } from '../xml/invoice-xml-partyType-dto'
import { Metadata } from 'src/app/shared/classes/metadata'
import { PortReadDto } from './port-read-dto'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface InvoiceReadDto extends Metadata {

    invoiceId: Guid
    date: string
    tripDate: string
    invoiceNo: number
    customer: SimpleEntity
    destination: SimpleEntity
    documentType: DocumentTypeVM
    paymentMethod: SimpleEntity
    ship: SimpleEntity
    aade: AadeVM
    issuer: InvoiceXmlPartyTypeDto
    counterPart: InvoiceXmlPartyTypeDto
    invoicesPorts: PortReadDto[]
    remarks: string
    isEmailSent: boolean
    isCancelled: boolean
    adults: number
    kids: number
    free: number
    totalPax: number
    netAmount: number
    vatPercent: number
    vatAmount: number
    grossAmount: number

}
